Source/GSThemeDrawing.m

```diff
@@ -2646,51 +2644,47 @@ - (void) drawScrollViewRect: (NSRect)rect
       NSScroller *vertScroller = [scrollView verticalScroller];
       NSScroller *horizScroller = [scrollView horizontalScroller];
       CGFloat scrollerWidth = [NSScroller scrollerWidth];
-      CGFloat borderWidth = [self sizeForBorderType: borderType].width;
+      CGFloat scrollerHeight;
 
       [color set];
 
       if ([scrollView hasVerticalScroller])
 	{
 	  NSInterfaceStyle style;
-	  CGFloat xpos;
-	  CGFloat scrollerHeight = bounds.size.height;
+	  CGFloat xpos = [vertScroller frame].origin.x;
+
+          scrollerHeight = [vertScroller frame].size.height;
 
 	  style = NSInterfaceStyleForKey(@"NSScrollViewInterfaceStyle", nil);
 	  if (style == NSMacintoshInterfaceStyle
 	      || style == NSWindows95InterfaceStyle)
 	    {
-              xpos = [vertScroller frame].origin.x - 1.0;
+              xpos -= 1.0;
 	    }
 	  else
 	    {
-              xpos = [vertScroller frame].origin.x + scrollerWidth;
+              xpos += scrollerWidth;
 	    }
           NSRectFill(NSMakeRect(xpos, [vertScroller frame].origin.y, 
-                                1.0, scrollerHeight - (borderWidth * 2)));
+                                1.0, scrollerHeight));
 	}
 
       if ([scrollView hasHorizontalScroller])
 	{
-	  CGFloat ypos;
-	  CGFloat scrollerY = [horizScroller frame].origin.y;
-	  CGFloat scrollerLength = bounds.size.width;
+	  CGFloat ypos = [horizScroller frame].origin.y;
 
-	  if ([scrollView hasVerticalScroller])
-	    {
-	      scrollerLength -= [NSScroller scrollerWidth];
-	    }
+          scrollerHeight = [horizScroller frame].size.width;
```

Comment:
This value definitely is not a "height". We need to be very careful with variable names. I would also suggest to keep variable definitions as local as possible. Move scrollerHeight into the other "if" branch and have a new local variable here (hScrollerWidth?). Perhaps we should also get the frame of the scroller only once in each branch? Then we would not need a local variable at all.

Reply:
As opposite to [NSScroller scrollerWidth] it would be logical to name it `scrollerHeight`. And it is definitely not "width" because it's confusing to method name. Width of horizontal scroller is visual height, while width of vertical scroller is visual width.
Anyway, I've made changes.